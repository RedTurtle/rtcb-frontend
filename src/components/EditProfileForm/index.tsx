import * as React from 'react';
import Select from 'react-select';

import { roles } from '../Giocatore';

type Roles = typeof roles;

interface IProps {
  player: {
    firstName: string;
    lastName: string;
    role: keyof Roles;
    versatile: boolean;
  };
}

interface IState {
  firstName: string;
  lastName: string;
  preferredRole: keyof Roles;
  versatile: boolean;
}

class EditProfileForm extends React.Component<IProps, IState> {
  public constructor(props: IProps) {
    super(props);
    const { firstName, lastName, role, versatile } = props.player;
    this.state = {
      firstName,
      lastName,
      preferredRole: role,
      versatile,
    };
  }

  public render() {
    return (
      <form>
        <div className="form-group">
          <label htmlFor="firstName">Nome</label>
          <input
            className="form-control"
            name="firstName"
            id="firstName"
            placeholder="Indica un nome..."
            value={this.state.firstName}
            onChange={this.onChangeFirstName}
          />
        </div>
        <div className="form-group">
          <label htmlFor="lastName">Cognome</label>
          <input
            className="form-control"
            name="lastName"
            id="lastName"
            placeholder="Indica un cognome..."
            value={this.state.lastName}
            onChange={this.onChangeLastName}
          />
        </div>
        <div className="form-group">
          <label htmlFor="preferred-role">Ruolo preferito</label>
          <Select
            name="prefrole"
            id="preferred-role"
            value={this.state.preferredRole}
            onChange={this.onChangeRole}
            clearable={false}
            options={Object.keys(roles).map(role => ({
              label: roles[role],
              value: role as string,
            }))}
          />
        </div>
        <div className="form-group">
          <label htmlFor="versatile">Versatile</label>
          <input
            name="versatile"
            id="versatile"
            type="checkbox"
            checked={this.state.versatile}
            onChange={this.onChangeVersatile}
          />
        </div>
        <div className="form-submit">
          <button
            type="submit"
            className="btn btn-primary"
            onClick={this.onSubmit}
          >
            Salva
          </button>
        </div>
      </form>
    );
  }

  private onChangeFirstName = (e: React.FormEvent<HTMLInputElement>): void => {
    e.preventDefault();
    const newName = e.currentTarget.value;
    this.setState(
      (prevState: IState): IState => ({
        ...prevState,
        firstName: newName,
      }),
    );
  };

  private onChangeLastName = (e: React.FormEvent<HTMLInputElement>): void => {
    e.preventDefault();
    const newName = e.currentTarget.value;
    this.setState(
      (prevState: IState): IState => ({
        ...prevState,
        lastName: newName,
      }),
    );
  };

  private onChangeRole = ({
    value,
  }: {
    value: IState['preferredRole'];
  }): void => {
    this.setState(
      (prevState: IState): IState => ({
        ...prevState,
        preferredRole: value,
      }),
    );
  };

  private onChangeVersatile = (e: React.FormEvent<HTMLInputElement>): void => {
    const newValue = e.currentTarget.checked;
    this.setState(
      (prevState: IState): IState => ({
        ...prevState,
        versatile: newValue,
      }),
    );
  };

  private onSubmit = (e: React.FormEvent<HTMLButtonElement>): void => {
    e.preventDefault();
  };
}

export default EditProfileForm;
