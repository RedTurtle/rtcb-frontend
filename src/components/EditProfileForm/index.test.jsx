import * as React from 'react';
import renderer from 'react-test-renderer';
import EditProfileForm from './index';

it('should match previous snapshot', () => {
  const component = renderer.create(
    <EditProfileForm
      player={{
        firstName: 'Ciccio',
        lastName: 'Pasticcio',
        role: 'D',
        versatile: false,
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
