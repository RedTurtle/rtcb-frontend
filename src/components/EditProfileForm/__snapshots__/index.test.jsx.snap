// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<form>
  <div
    className="form-group"
  >
    <label
      htmlFor="firstName"
    >
      Nome
    </label>
    <input
      className="form-control"
      id="firstName"
      name="firstName"
      onChange={[Function]}
      placeholder="Indica un nome..."
      value="Ciccio"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="lastName"
    >
      Cognome
    </label>
    <input
      className="form-control"
      id="lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Indica un cognome..."
      value="Pasticcio"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="preferred-role"
    >
      Ruolo preferito
    </label>
    <div
      className="Select has-value is-searchable Select--single"
      style={undefined}
    >
      <input
        disabled={false}
        name="prefrole"
        type="hidden"
        value="D"
      />
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-value"
            style={undefined}
            title={undefined}
          >
            <span
              aria-selected="true"
              className="Select-value-label"
              id="react-select-2--value-item"
              role="option"
            >
              Difensore
            </span>
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id="preferred-role"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="versatile"
    >
      Versatile
    </label>
    <input
      checked={false}
      id="versatile"
      name="versatile"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="form-submit"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="submit"
    >
      Salva
    </button>
  </div>
</form>
`;
